src/interpolation_utils.jl

```diff
@@ -54,11 +55,21 @@ function munge_data(U::StridedMatrix, t::AbstractVector)
     TU = Base.nonmissingtype(eltype(U))
     Tt = Base.nonmissingtype(eltype(t))
     @assert length(t) == size(U, 2)
-    non_missing_indices = collect(i
-    for i in 1:length(t)
-    if !any(ismissing, U[:, i]) && !ismissing(t[i]))
-    newUs = [TU.(U[:, i]) for i in non_missing_indices]
-    newt = Tt.([t[i] for i in non_missing_indices])
+    non_missing_indices = collect(
+        idx for idx in 1:length(t)
+    if !any(ismissing, U[:, idx]) && !ismissing(t[idx])
+    )
+    newUs = [TU.(U[:, idx]) for idx in non_missing_indices]
+    newt = Tt.([t[idx] for idx in non_missing_indices])
 
     return hcat(newUs...), newt
 end
+
+function get_idx(tvec, t, iguess; lb = 1, ub_shift = -1, idx_shift = 0, side = :last)
+    ub = length(tvec) + ub_shift
+    return if side == :last
+        clamp(searchsortedlastcorrelated(tvec, t, iguess) + idx_shift, lb, ub)
+    else
+        clamp(searchsortedfirstcorrelated(tvec, t, iguess) + idx_shift, lb, ub)
+    end
```

Comment:
And an else condition to `error(...)`. This is to ensure it goes through the right code path.